File: Source/Chargify.Web/App_Code/SimpleLog.vb
#Region "Author Note"
' SimpleLog.aspx
' Author: Kori Francis <twitter.com/djbyter>
#End Region

Imports Microsoft.VisualBasic
Imports System.IO

''' <summary>
''' SimpleLog just writes a text log file, and writes the time for every entry.
''' </summary>
Public Class SimpleLog
    Private sLogFormat As String
    Private sErrorTime As String

    Public Sub New()
        sLogFormat = DateTime.Now.ToShortDateString().ToString & " " & DateTime.Now.ToLongTimeString().ToString() & " ==> "
        Dim sYear As String = Date.Now.Year.ToString("0000")
        Dim sMonth As String = Date.Now.Month.ToString("00")
        Dim sDay As String = Date.Now.Day.ToString("00")
        sErrorTime = sYear & sMonth & sDay
    End Sub

    Public Sub Write(ByVal sPathName As String, ByVal sErrorMsg As String)
        Dim sw As New StreamWriter(sPathName & sErrorTime & ".txt", True)
        sw.WriteLine(sLogFormat & sErrorMsg)
        sw.Flush()
        sw.Close()
    End Sub
End Class

File: Source/ChargifyConsoleSample/SimpleLog.cs
using System;
using System.IO;

namespace ChargifyConsoleSample
{
    /// <summary>
    /// SimpleLog just writes a text log file, and writes the time for every entry.
    /// </summary>
    public class SimpleLog
    {
        private string sLogFormat;

        private string sErrorTime;
        public SimpleLog()
        {
            sLogFormat = DateTime.Now.ToShortDateString() + " " + DateTime.Now.ToLongTimeString() + " ==> ";
            string sYear = System.DateTime.Now.Year.ToString("0000");
            string sMonth = System.DateTime.Now.Month.ToString("00");
            string sDay = System.DateTime.Now.Day.ToString("00");
            sErrorTime = sYear + sMonth + sDay;
        }

        public void Write(string sPathName, string sErrorMsg, bool useLogFormat)
        {
            StreamWriter sw = new StreamWriter(sPathName + sErrorTime + ".txt", true);
            if (useLogFormat)
            {
                sw.WriteLine(sLogFormat + sErrorMsg);
            }
            else
            {
                sw.WriteLine(sErrorMsg);
            }
            sw.Flush();
            sw.Close();
        }
    }
}
